manager/src/main/java/com/redhat/service/smartevents/manager/models/Processor.java

```diff
@@ -32,7 +32,7 @@
         @NamedQuery(name = "PROCESSOR.findByBridgeIdAndName",
                 query = "from Processor p where p.name=:name and p.bridge.id=:bridgeId"),
         @NamedQuery(name = "PROCESSOR.findByShardIdWithReadyDependencies",
-                query = "select p " +
+                query = "select distinct(p) " +
```

Comment:
I'm not sure we need the `distinct(p)` addition.

It is probably better to remove the join on `ConnectorEntities` (as the table alias `c` seems to be unused).

I suspect the join was required at [some point](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/commit/ddbf451ab0cb812a5e4b0759afade4e2160c3ba4#diff-d1d6ed99f75d8cbbe6d58d663083116f0942c57e8db75b7bd6bf9ede7da6dabeR42) to check MC's status (but this is now part of the `Worker` mechanism).